### PR TITLE
Making publisher constructor call async and refactoring code to call dispose only once.

### PR DESCRIPTION
**Pull request by khilangoel**
Added ETW event for dispose method also.

### CODE REVIEW

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs [hunk 1]
```diff
@@ -66,17 +69,24 @@ public TestRequestManager()
...
                 TestRunResultAggregator.Instance,
-                TestPlatformEventSource.Instance)
+                TestPlatformEventSource.Instance,
+                MetricsPublisherFactory.GetMetricsPublisher(IsTelemetryOptedIn(), CommandLineOptions.Instance.IsDesignMode))
+        {
+        }
+
+        public void Dispose()
         {
+            this.metricsPublisher.Result.Dispose();
```
Comment (navin22) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: Add the regular pattern of handling the Dispose with one more method `Dispose(bool disposing)`

Comment (navin22) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: null check required ?

Comment (khilan2goel) [reply src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: Should I add destructor as well in that case?

**Review by navin22 — COMMENTED** [review]
LGTM

**Comment by khilan2goel** [comment]
test Windows_NT / Release Build

**Review by harshjain2 — APPROVED** [review]

**Pull request merged**